src/Controllers/controller.py

```diff
@@ -2,18 +2,59 @@
 from Controllers.cli_controller import CLI_Controller
 from Controllers.gui_controller import GUI_Controller
 from Views.cli_view import CLI_View
+from prompt_toolkit import prompt
+from prompt_toolkit.completion import NestedCompleter
 class UML_Controller:
 
     def __init__(self):
+        """
+        Initializes the instance with either a CLI_Controller or GUI_Controller based on user choice, 
+        and sets up autocomplete functionality.
+        """
         self._controller:CLI_Controller | GUI_Controller = self.__pick_controller()
+        self.setup_autocomplete()  
```

Comment:
does the gui have autocomplete? If it doesn't, this should be in the CLI Controller instead of the decorator.

Reply:
The GUI doesn't have autocomplete but the way the autocomplete works is that it replaces the input command with prompt, and based on the current codebase, we call input from the controller